### ISSUE TITLE
TestDirectOutgoing_AdapterSendsUnicast is flaky

### ISSUE THREAD

**Issue by electricmonk** [orbs-network/orbs-network-go#894]
https://circleci.com/gh/orbs-network/orbs-network-go/10241
```
Failed
t.go:22: �[32minfo 2019-02-14T08:13:56.457176Z attempting outgoing transport connection adapter=gossip server=127.0.0.1:31020 entry-point=Gossip.Transport.TCP.Client.127.0.0.1:31020 request-id=Gossip.Transport.TCP.Client.127.0.0.1:31020-1550132036457155607 function=tcp.(*directTransport).clientMainLoop source=services/gossip/adapter/tcp/direct_outgoing.go:18 
t.go:22: �[36minfo 2019-02-14T08:13:56.457181Z attempting outgoing transport connection adapter=gossip server=127.0.0.1:31021 entry-point=Gossip.Transport.TCP.Client.127.0.0.1:31021 request-id=Gossip.Transport.TCP.Client.127.0.0.1:31021-1550132036457166363 function=tcp.(*directTransport).clientMainLoop source=services/gossip/adapter/tcp/direct_outgoing.go:18 
t.go:22: info 2019-02-14T08:13:56.457203Z gossip transport server listening adapter=gossip port=44917 function=tcp.(*directTransport).serverMainLoop source=services/gossip/adapter/tcp/direct_incoming.go:53 
t.go:22: �[32minfo 2019-02-14T08:13:56.457267Z successful outgoing gossip transport connection adapter=gossip peer=127.0.0.1:31020 entry-point=Gossip.Transport.TCP.Client.127.0.0.1:31020 request-id=Gossip.Transport.TCP.Client.127.0.0.1:31020-1550132036457155607 function=tcp.(*directTransport).clientHandleOutgoingConnection source=services/gossip/adapter/tcp/direct_outgoing.go:35 
t.go:22: �[36minfo 2019-02-14T08:13:56.457278Z successful outgoing gossip transport connection adapter=gossip peer=127.0.0.1:31021 entry-point=Gossip.Transport.TCP.Client.127.0.0.1:31021 request-id=Gossip.Transport.TCP.Client.127.0.0.1:31021-1550132036457166363 function=tcp.(*directTransport).clientHandleOutgoingConnection source=services/gossip/adapter/tcp/direct_outgoing.go:35 
t.go:22: �[35minfo 2019-02-14T08:13:56.482492Z successful incoming gossip transport connection adapter=gossip peer=127.0.0.1:45360 entry-point=Gossip.Transport.TCP.Server request-id=Gossip.Transport.TCP.Server-1550132036457219742 function=tcp.(*directTransport).serverHandleIncomingConnection source=services/gossip/adapter/tcp/direct_incoming.go:79 
require.go:157: 
Error Trace:	direct_outgoing_test.go:81
            				context.go:11
            				direct_outgoing_test.go:66
Error:      	Not equal: 
            	expected: []byte{0x2, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x11, 0x0, 0x0, 0x0, 0x2, 0x0, 0x0, 0x0, 0x22, 0x33, 0x0, 0x0}
            	actual  : []byte{0x0, 0x0, 0x0, 0x0, 0x2, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x11, 0x0, 0x0, 0x0, 0x2, 0x0, 0x0, 0x0}
            	
            	Diff:
            	--- Expected
            	+++ Actual
            	@@ -1,4 +1,4 @@
            	 ([]uint8) (len=20) {
            	- 00000000  02 00 00 00 01 00 00 00  11 00 00 00 02 00 00 00  |................|
            	- 00000010  22 33 00 00                                       |"3..|
            	+ 00000000  00 00 00 00 02 00 00 00  01 00 00 00 11 00 00 00  |................|
            	+ 00000010  02 00 00 00                                       |....|
            	 }
Test:       	TestDirectOutgoing_AdapterSendsUnicast
t.go:22: �[35minfo 2019-02-14T08:13:56.482999Z failed receiving transport data, disconnecting adapter=gossip error=EOF peer=127.0.0.1:45360 entry-point=Gossip.Transport.TCP.Server request-id=Gossip.Transport.TCP.Server-1550132036457219742 function=tcp.(*directTransport).serverHandleIncomingConnection source=services/gossip/adapter/tcp/direct_incoming.go:87 
t.go:22: �[36minfo 2019-02-14T08:13:56.483032Z client loop stopped since server is shutting down adapter=gossip entry-point=Gossip.Transport.TCP.Client.127.0.0.1:31021 request-id=Gossip.Transport.TCP.Client.127.0.0.1:31021-1550132036457166363 function=tcp.(*directTransport).clientHandleOutgoingConnection source=services/gossip/adapter/tcp/direct_outgoing.go:73 
t.go:22: �[32minfo 2019-02-14T08:13:56.483064Z incoming connection accept stopped since server is shutting down adapter=gossip entry-point=Gossip.Transport.TCP.Server request-id=Gossip.Transport.TCP.Server-1550132036482445589 function=tcp.(*directTransport).serverMainLoop source=services/gossip/adapter/tcp/direct_incoming.go:65 
t.go:22: �[32minfo 2019-02-14T08:13:56.483045Z client loop stopped since server is shutting down adapter=gossip entry-point=Gossip.Transport.TCP.Client.127.0.0.1:31020 request-id=Gossip.Transport.TCP.Client.127.0.0.1:31020-1550132036457155607 function=tcp.(*directTransport).clientHandleOutgoingConnection source=services/gossip/adapter/tcp/direct_outgoing.go:73 
```

**Comment by electricmonk** [comment]
duplicates #973

**Comment by ronnno** [comment]
Transmission is shifted by 4 bytes. presumably a keepalive message.

this issue was resolved in this commit: 90eaa44fb4e7ceb475f7533b9d7c2bc52d281e7f
